### DISCUSSION TITLE
Add support for CLI command extensions

### DISCUSSION THREAD

**Discussion by WoLfulus** [directus/directus#3582] [Feature Requests]
When enhancing Directus experience through packages and/or building custom installations, it would be nice to have support for adding new CLI commands to Directus through extensions/packages. 

Same as structure as `endpoints` or `hooks`, but for CLI commands, or even a hook to register new commands at any level (even root level like `directus ext --arg`) as IMHO allowing root level commands are cleaner. 

If root level commands are not possible (due to possible name clashes), it would be nice to have them under `directus ext <command>` or `directus extension <command>` instead of `directus custom <command>`

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
